### PR TITLE
feature: update gitlab and quicksight visualisation last updated dates

### PR DESCRIPTION
**Pull request by abbas123456**
https://trello.com/c/NgNgdFFl/945-spike-data-visualisation-last-updated-information

### Checklist

* [ ] Have tests been added to cover any changes?
* [ ] Has the [CHANGELOG](https://github.com/uktrade/data-workspace/blob/master/CHANGELOG.md) been updated?

### CODE REVIEW

File: dataworkspace/dataworkspace/apps/datasets/views.py [hunk 1]
```diff
@@ -338,7 +340,7 @@ def sorted_datasets_and_visualisations_matching_query_for_user(query, use, user)
     all_datasets = (
         master_and_datacut_datasets.union(reference_datasets)
         .union(visualisations)
-        .order_by('-search_rank', 'name')
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/apps/datasets/views.py:341]: Has this just slipped in? Doesn't feel like part of this change.

Comment (abbas123456) [reply dataworkspace/dataworkspace/apps/datasets/views.py:341]: I have changed the order_by clause to be '-modified_date' first so I can confirm that the catalogue items are getting updated in the celery task without having to check the sql database

File: dataworkspace/dataworkspace/apps/applications/utils.py [hunk 2]
```diff
@@ -907,3 +910,47 @@ def sync_quicksight_permissions(
...
+        latest_commit = master_branch['commit']
+        latest_commit_date = datetime.datetime.strptime(
+            latest_commit['committed_date'], '%Y-%m-%dT%H:%M:%S.%f%z'
+        )
+
+        visualisation.modified_date = latest_commit_date
+        visualisation.save()
+
+    # AWS QuickSight visualisations
+    for visualisation in VisualisationCatalogueItem.objects.filter(visualisation_template__isnull=True):
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/apps/applications/utils.py:935]: I think it's technically possible for a catalogue item to have a gitlab visualisation and a quicksight visualisation, so this is probably not quite the right query.

Comment (abbas123456) [reply dataworkspace/dataworkspace/apps/applications/utils.py:935]: From Alex:

The data visualisation add/edit page on Django admin supports both types but I'm not aware of a use case for including both and don't think there are any instances of this.

Comment (samuelhwilliams) [reply dataworkspace/dataworkspace/apps/applications/utils.py:935]: Yeah I agree, there's not much of a use case ... but it's technically possible, and presumably also fairly trivial to update the query to support it, so I guess my question would be "why not?"

File: dataworkspace/dataworkspace/apps/datasets/views.py (hunk 1 above)
Comment (samuelhwilliams) [reply dataworkspace/dataworkspace/apps/datasets/views.py:341]: Yep, I understand why you've done it of course - just saying that it's probably not a change we'd want to merge as part of this?

Comment (abbas123456) [reply dataworkspace/dataworkspace/apps/datasets/views.py:341]: Oh ok sorry, yeah the plan was to remove it after I've finished testing

File: dataworkspace/dataworkspace/apps/applications/utils.py (hunk 2 above)
Comment (abbas123456) [reply dataworkspace/dataworkspace/apps/applications/utils.py:935]: Yeah fair enough, will update it

**Comment by abbas123456** [comment]
Will close the PR for now as have decided to focuson making a catalogue item's "Published date" available as an option for sorting first.

**Pull request closed without merging**